plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveFileFormats.java

```diff
@@ -182,7 +184,9 @@ public void testCsvFile(int rowCount, long fileSizePadding)
                 .withColumns(testColumns)
                 .withRowsCount(rowCount)
                 .withFileSizePadding(fileSizePadding)
-                .isReadableByRecordCursor(createGenericHiveRecordCursorProvider(HDFS_ENVIRONMENT));
+                .withFileWriterFactory(new CsvFileWriterFactory(HDFS_FILE_SYSTEM_FACTORY, TESTING_TYPE_MANAGER))
```

Comment:
This changes the test so that we're no longer testing against files written by Hive. I was looking at this today for Avro and was thinking we could change the `assertRead()` method to test with `fileWriterFactory` if present and without it. Then we don't need to copy all of the tests.

Reply:
I changed the test framework in the last commit to always test both generic writer and the supplied writer